### DISCUSSION TITLE
How to automate testing

### DISCUSSION THREAD

**Discussion by J0ergL** [mobilityhouse/ocpp#567] [Q&A]
Dear all,

Thank you very much for this python package!
I am programming a chargepoint using the package. So far I have done the testing manually but I d like to automate it.
Any hints and directions on how to accomplish this are more then welcome and are highly appreciated. 

regards

Jörg

**Comment by benoit-bremaud** [comment]
Dear Jörg,

I came across your message regarding the use of the python package for programming a chargepoint, and your interest in transitioning from manual to automated testing caught my attention. I'm in a similar situation and am exploring ways to automate Charging Station Test procedures. It's a journey that seems both challenging and exciting.

From my exploration and the resources I've come across, automating test procedures can significantly streamline the validation of charge point operations and OCPP functionalities. I'm currently looking into the Open Charge Point Protocol (OCPP) and how it can be leveraged for automated testing, including the use of simulation tools and frameworks designed for OCPP-compliant charging stations.

I would be thrilled to exchange ideas, insights, and resources on this topic. If I find any useful tools or methodologies, I'll be sure to share them with the community. It would be great if we could collaborate or share findings that could benefit others working on similar projects.

In the meantime, one approach could involve utilizing existing OCPP simulation tools that can mimic various charging scenarios and test the charge point's responses to different requests and conditions. These tools can be integrated into a CI/CD pipeline for regular testing as part of the development process. Additionally, exploring the use of unit testing frameworks to test individual functions and modules within the charge point software could be beneficial.

It's also important to ensure that the testing framework supports the specific version of OCPP that your chargepoint implements, whether it's 1.6, 2.0, or any other version, as the protocol features and capabilities vary between versions.

Feel free to reach out if you want to discuss this further or have any insights to share. Let's keep the community thriving by sharing knowledge and supporting each other's efforts in advancing EV charging technology.

Best regards,

Benoit

**Reply by J0ergL** [reply]
Dear Benoit,

Thank you for your reply and your suggestions. I am new to CI/CD and you read the signs well. It is exactly what I am aiming at. Currently it is only a mind game trying to figure out which approach is faster to the goal. I like the idea of using existing simulators. Although I was considering writing the CS with this OCPP library but I got stuck at the point on how to put the tests and the programmed CS together for automated testing. 
Yes I would like to discuss this further. Which real time channels to you suggest?
Do you know garther town?

Best regards,


Jörg

**Comment by P33rCharge** [comment]
Dear Benoit,

any progress on your side?
I had some other issues to work on in the mean time and now I am refocusing on the automated testing for CI/CD....

Regards Jörg
